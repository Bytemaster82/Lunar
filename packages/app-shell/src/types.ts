export type Context = {
  addPageData: (data: object, customID?: string) => string;
  addInfoToast: (message: string, props?: object) => string;
  addFailureToast: (message: string, props?: object) => string;
  addSuccessToast: (message: string, props?: object) => string;
  addRefreshToast: (message: string, props?: object) => string;
  data: { [key: string]: object };
  name: string;
  removePageData: (id: string) => void;
  removeToast: (id: string) => void;
  toasts: Toast[];
};

export type ToastType = 'info' | 'success' | 'danger' | 'refresh';

export type Toast = {
  id: string;
  message: string | Error;
  props?: object;
};
