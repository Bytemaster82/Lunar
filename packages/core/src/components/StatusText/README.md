Default and status classified text.

```jsx
<StatusText>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</StatusText>
<StatusText notice>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</StatusText>
<StatusText info>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</StatusText>
<StatusText success>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</StatusText>
<StatusText warning>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</StatusText>
<StatusText danger>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</StatusText>
<StatusText muted>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</StatusText>
```

Can pass props to the underlying `Text` component.

```jsx
<StatusText notice micro>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</StatusText>
<StatusText info small>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</StatusText>
<StatusText success large>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</StatusText>
<StatusText warning disabled>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</StatusText>
<StatusText danger light>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</StatusText>
<StatusText muted bold>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</StatusText>
```
