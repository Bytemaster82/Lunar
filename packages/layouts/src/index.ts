// Empty on purpose to resolve TypeScript declarations
