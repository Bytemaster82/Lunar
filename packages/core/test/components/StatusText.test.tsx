import React from 'react';
import { shallow } from 'enzyme';
import StatusText from '../../src/components/StatusText';
import { STATUSES } from '../../src/constants';

describe('<StatusText />', () => {
  it('errors when multiple states are used at once', () => {
    expect(() => {
      shallow(
        <StatusText danger success>
          Default
        </StatusText>,
      ).dive();
    }).toThrowError();
  });

  describe('statuses', () => {
    STATUSES.forEach(status => {
      it('renders text', () => {
        const wrapper = shallow(<StatusText {...{ [status]: true }}>{status}</StatusText>)
          .dive()
          .dive()
          .dive();

        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  it('can pass props to underlying `Text`', () => {
    const wrapper = shallow(
      <StatusText large muted>
        Text
      </StatusText>,
    )
      .dive()
      .dive()
      .dive();

    expect(wrapper).toMatchSnapshot();
  });
});
