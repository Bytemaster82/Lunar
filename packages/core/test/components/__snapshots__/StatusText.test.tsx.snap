// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusText /> can pass props to underlying \`Text\` 1`] = `
<h4
  className="text_i23n2q-o_O-text_large_1b2mxt3"
>
  <span
    className="text_muted_10k87om"
  >
    Text
  </span>
</h4>
`;

exports[`<StatusText /> statuses renders text 1`] = `
<div
  className="text_i23n2q"
>
  <span
    className="text_notice_16fnaxc"
  >
    notice
  </span>
</div>
`;

exports[`<StatusText /> statuses renders text 2`] = `
<div
  className="text_i23n2q"
>
  <span
    className="text_info_1qlg6bk"
  >
    info
  </span>
</div>
`;

exports[`<StatusText /> statuses renders text 3`] = `
<div
  className="text_i23n2q"
>
  <span
    className="text_success_xr5ae0"
  >
    success
  </span>
</div>
`;

exports[`<StatusText /> statuses renders text 4`] = `
<div
  className="text_i23n2q"
>
  <span
    className="text_warning_1b6zul9"
  >
    warning
  </span>
</div>
`;

exports[`<StatusText /> statuses renders text 5`] = `
<div
  className="text_i23n2q"
>
  <span
    className="text_danger_1ardfxt"
  >
    danger
  </span>
</div>
`;

exports[`<StatusText /> statuses renders text 6`] = `
<div
  className="text_i23n2q"
>
  <span
    className="text_muted_10k87om"
  >
    muted
  </span>
</div>
`;
